app/kuma-dp/pkg/config/validate.go

```diff
@@ -26,10 +29,36 @@ func ValidateTokenPath(path string) error {
 		return errors.Wrapf(err, "could not read the token in the file %s", path)
 	}
 
-	_, _, err = new(jwt.Parser).ParseUnverified(string(rawToken), &jwt.MapClaims{})
+	token, parts, err := new(jwt.Parser).ParseUnverified(string(rawToken), &jwt.MapClaims{})
 	if err != nil {
-		return errors.Wrapf(err, "token in the file %s is not valid JWT token. Double check blank characters in the file", path)
+		return errors.Wrapf(err, "token in the file %s is not valid JWT token. Double check for blank characters in the file", path)
+	}
+
+	core.Log.Info("dump ", "token", token, "parts ", parts)
+	if token.Method.Alg() == "" {
+		return errors.New("not valid JWT token. No Alg.")
+	}
+
+	if token.Header == nil {
+		return errors.New("not valid JWT token. No Header.")
+	}
+	if len(parts) != 3 {
+		return errors.New("not valid JWT token. Must have 3 parts..")
+	}
+	for _, part := range parts {
+		if !isASCII(part) {
+			return errors.New("token in the file %s is not valid JWT token. Found a non-printable characters in the token.")
```

Comment:
1) Can we have a test with `\n` in the token?
2) If I have `\n` in the file will this go to this error or the previous one? Either way. The error should also say something like `The file cannot have blank characters like empty lines. Example how to get rid of non-printable characters: "tr -cd '[:print:]' < token > token"`